### PR TITLE
[awss3receiver] Mark the AWS S3 Receiver as Alpha and enable in for otelcontribcol

### LINKED ISSUE #30750: New component: AWS S3 Receiver
### The purpose and use-cases of the new component

The S3 receiver will allow the retrieval and processing of telemetry data previously stored in S3 by the AWS S3 Exporter.
This will make it possible to retrieve data previously cold stored in S3 and allow us to investigate issues not reported within the time span data is available in our Observability service provider.

### Example configuration for the component
```yaml
receivers:
  awss3:
    s3downloader:
      s3_bucket: abucket
      s3_prefix: tenant_a
      s3_partition: minute
    starttime: "2024-01-13 15:00"
    endtime: "2024-01-21 15:00"
```
### Telemetry data types supported

- traces
- metrics
- logs

### Is this a vendor-specific component?

- [ ] This is a vendor-specific component
- [ ] If this is a vendor-specific component, I am proposing to contribute and support it as a representative of the vendor.

### Code Owner(s)

adcharre

### Sponsor (optional)

@atoulme
### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adcharre**
#### Description
Mark awss3receiver as alpha and enable it in the otelcontribcol

#### Link to tracking issue
Closes #30750

### CODE REVIEW

File: cmd/otelcontribcol/builder-config.yaml [hunk 1]
```diff
@@ -136,6 +136,7 @@ receivers:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver v0.111.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsecscontainermetricsreceiver v0.111.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsfirehosereceiver v0.111.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver v0.111.0
```
Comment (songy23) [inline cmd/otelcontribcol/builder-config.yaml:139]: FYI you may want to update https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml as well

**Comment by songy23** [comment]
> Generated code is out of date, please run "make gendistributions" and commit the changes in this PR.

**Review by atoulme — APPROVED** [review]

**Comment by adcharre** [comment]
> > Generated code is out of date, please run "make gendistributions" and commit the changes in this PR.

Done.

**Review by songy23 — APPROVED** [review]

**Comment by songy23** [comment]
Please rebase

**Comment by adcharre** [comment]
> Please rebase

Done

**Pull request merged**